Runtime/Triangulator.cs

```diff
@@ -1098,26 +1082,18 @@ public void Execute()
                     pointToHalfedge[triangles[i]] = i;
                 }
 
-                BuildInternalConstraints();
-
-                foreach (var c in internalConstraints)
-                {
-                    TryApplyConstraint(c);
-                }
-            }
-
-            private void BuildInternalConstraints()
-            {
-                for (int index = 0; index < internalConstraints.Length; index++)
+                for (int index = 0; index < inputConstraintEdges.Length / 2; index++)
                 {
-                    internalConstraints[index] = new(
-                        idA: inputConstraintEdges[2 * index + 0],
-                        idB: inputConstraintEdges[2 * index + 1]
+                    var c = math.int2(
```

Comment:
Imo using an int2 for an edge is not excellent. I much prefer a custom struct, so that it's clear that we are accessing edges, not some kind of coordinate.
LLVM can also sometimes do suboptimal things with these, as they are often mapped to SIMD-registers, but the code doesn't use them in a SIMD-like way.

Reply:
> Imo using an int2 for an edge is not excellent. I much prefer a custom struct, so that it's clear that we are accessing edges, not some kind of coordinate.
LLVM can also sometimes do suboptimal things with these, as they are often mapped to SIMD-registers, but the code doesn't use them in a SIMD-like way.

The `Edge` struct was introduced early in the project when there were mappings like  `EdgeToTriangle`/`TriangleToEdge` etc. However, I'm not a fan of having an almost empty struct for this purpose. I prefer to keep this project as simple as possible. We could use pair of int `int ci; int cj` or tuple `(int ci, int cj)` since this doesn't exit the bursted code. However, I believe that in the context of usage, `int2` should be safe enough.